model/DataObject.php

```diff
@@ -1744,44 +1745,79 @@ public function getManyManyComponents($componentName, $filter = "", $sort = "",
 	}
 
 	/**
+	 * @deprecated 4.0 Method has been renamed to hasOne()
+	 * @param string $component
+	 * @return array|null
+	 */
+	public function has_one($component = null) {
+		Deprecation::notice('4.0', 'Method has been renamed to hasOne()');
+		return $this->hasOne($component);
+	}
+
+	/**
 	 * Return the class of a one-to-one component.  If $component is null, return all of the one-to-one components and
 	 * their classes. If the selected has_one is a polymorphic field then 'DataObject' will be returned for the type.
 	 *
 	 * @param string $component Name of component
-	 *
 	 * @return string|array The class of the one-to-one component, or an array of all one-to-one components and their
 	 *                      classes.
 	 */
-	public function has_one($component = null) {
-		$classes = ClassInfo::ancestry($this);
+	public function hasOne($component = null) {
+		if($component) {
+			return $this->hasOneComponent($component);
+		}
 
+		$classes = ClassInfo::ancestry($this);
 		foreach($classes as $class) {
 			// Wait until after we reach DataObject
 			if(in_array($class, array('Object', 'ViewableData', 'DataObject'))) continue;
 
-			if($component) {
-				$hasOne = Config::inst()->get($class, 'has_one', Config::UNINHERITED);
-
-				if(isset($hasOne[$component])) {
-					return $hasOne[$component];
+			$newItems = (array)Config::inst()->get($class, 'has_one', Config::UNINHERITED);
+			// Validate the data
+			foreach($newItems as $k => $v) {
+				if(!is_string($k) || is_numeric($k) || !is_string($v)) {
+					user_error("$class::\$has_one has a bad entry: "
```

Comment:
Just keeping the existing behaviour. I think `user_error` makes sense anyway - would you ever actually want to catch this and try to carry on?